extensions/analyticsdx-vscode-core/src/commands/gatherers/outputDirGatherer.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as glob from 'glob';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { nls } from '../../messages';
+import { getRootWorkspacePath } from '../../util/rootWorkspace';
+import { getPackageDirectoryPaths } from '../../util/sfdx';
+import { CancelResponse, ContinueResponse, ParametersGatherer } from '../commands';
+
+export type OutputDirType = { outputDir: string };
+
+/** Prompts to pick a valid folder in the sfdx packge directories in the workspace for file creation.
+ * The returned output folder will use platform-native seperators.
+ */
+export class OutputDirGatherer implements ParametersGatherer<OutputDirType> {
```

Comment:
This is used to pick the directory (when called from the command palette). It works the same as `SFDX: Create Lightning Web Component` from the salesforcedx extension.